crates/ika-core/src/dwallet_mpc/dwallet_mpc_service.rs

```diff
@@ -183,69 +217,137 @@ impl DWalletMPCService {
                     .await;
             }
 
-            self.dwallet_mpc_manager
-                .handle_dwallet_db_message(DWalletMPCDBMessage::PerformCryptographicComputations)
-                .await;
-        }
-    }
-
-    /// Receive all completed MPC sessions from the MPC Output Verifier over the
-    /// `consensus_round_completed_sessions` channel.
-    /// If the session exists, mark is as [`MPCSessionStatus::Finished`].
-    /// Otherwise, create a new session with that status, to avoid re-running the computation for it.
-    fn receive_completed_mpc_session_identifiers(&mut self, bootstrap: bool) {
-        let mut completed_sessions = HashSet::new();
-        loop {
-            match self.consensus_round_completed_sessions_receiver.try_recv() {
-                Err(mpsc::error::TryRecvError::Empty) => {
-                    // No more completed sessions to report at the moment.
-                    break;
+            for (round, outputs) in mpc_outputs {
+                let mut messages = vec![];
+                let mut completed_sessions = vec![];
+                for output in outputs {
+                    let output_result = self
+                        .dwallet_mpc_manager
+                        .handle_dwallet_db_output(&output)
+                        .await;
+                    match output_result {
+                        Ok(output_result) => match output_result.result {
+                            OutputVerificationStatus::FirstQuorumReached(m) => {
+                                messages.extend(m);
```

Comment:
log the round number we reached this output, and the hash on contents, and the session identifier

Reply:
try_verify_output is only called once and the result go back to manager to service, I will add logs there